Classes/Http/SetJwtCookieComponent.php

```diff
@@ -78,13 +87,38 @@ private function isOpenIdConnectAuthentication(): bool
         return false;
     }
 
+    /**
+     * @param ComponentContext $componentContext
+     * @return bool
+     */
+    private function isLogoutRequest(ComponentContext $componentContext): bool
+    {
+        $httpRequest = $componentContext->getHttpRequest();
+        if (!$httpRequest instanceof ServerRequestInterface) {
+            return false;
+        }
+        $queryParams = $httpRequest->getQueryParams();
+        return isset($queryParams['logout']) && $queryParams['logout'] === $this->options['serviceName'];
+    }
+
+    /**
+     * @param ComponentContext $componentContext
+     * @return void
+     */
+    private function handleLogout(ComponentContext $componentContext): void
+    {
+        $this->removeJwtCookie($componentContext);
+        $this->logger->info(sprintf('OpenID Connect Client: (%s) Logout requested (via query parameter) removing JWT cookie for service "%s".', get_class($this), $this->options['serviceName']));
+        $componentContext->replaceHttpResponse($componentContext->getHttpResponse()->withHeader('Location', (string)$componentContext->getHttpRequest()->getUri()->withQuery('')));
```

Comment:
This part is a bit magic.. It redirects to the current URI without query params. If other params are required they would be missing. I think we have to revise this part, but without the redirect the account is already authenticated

Reply:
What about a defined "afterLogout" URL?